SimPEG/regularization/rotated.py

```diff
@@ -247,3 +304,224 @@ def W(self):
+        the regularization based on the derivative of the l2-norm measure. And it assists the
+        convergence by ensuring the model does not deviate
+        aggressively from the global 2-norm solution during the first few IRLS iterations.
+        For a comprehensive description, see the documentation for :py:meth:`get_lp_weights` .
+
+        Returns
+        -------
+        bool
+            Whether to scale IRLS weights.
+        """
+        return self._irls_scaled
+
+    @irls_scaled.setter
+    def irls_scaled(self, value: bool):
+        self._irls_scaled = validate_type("irls_scaled", value, bool, cast=False)
+
+    @property
+    def irls_threshold(self):
+        r"""Stability constant for computing IRLS weights.
+
+        Returns
+        -------
+        float
+            Stability constant for computing IRLS weights.
+        """
+        return self._irls_threshold
+
+    @irls_threshold.setter
+    def irls_threshold(self, value):
+        self._irls_threshold = validate_float(
+            "irls_threshold", value, min_val=0.0, inclusive_min=False
+        )
+
+    @property
+    def norm(self):
+        r"""Norm for the sparse regularization.
+
+        Returns
+        -------
+        None, float, (n_cells, ) numpy.ndarray
+            Norm for the sparse regularization. If ``None``, a 2-norm is used.
+            A float within the interval [0,2] represents a constant norm applied for all cells.
+            A ``numpy.ndarray`` object, where each entry is used to apply a different norm to each cell in the mesh.
+        """
+        return self._norm
+
+    @norm.setter
+    def norm(self, value: float | np.ndarray | None):
+        if value is None:
+            value = np.ones(self.cell_gradient.shape[0]) * 2.0
+        else:
+            value = np.ones(self.cell_gradient.shape[0]) * value
+        if np.any(value < 0) or np.any(value > 2):
+            raise ValueError(
+                "Value provided for 'norm' should be in the interval [0, 2]"
+            )
+        self._norm = value
+
+    @property
+    def units(self) -> str | None:
+        """Units for the model parameters.
+
+        Some regularization classes behave differently depending on the units; e.g. 'radian'.
+
+        Returns
+        -------
+        str
+            Units for the model parameters.
+        """
+        return self._units
+
+    @units.setter
+    def units(self, units: str | None):
+        if units is not None and not isinstance(units, str):
+            raise TypeError(
+                f"'units' must be None or type str. Value of type {type(units)} provided."
+            )
+        self._units = units
+
+    @property
+    def _cell_distances(self) -> np.ndarray:
+        """
+        cell size average on faces
+
+        :return: np.ndarray
+        """
+        cell_distances = self.cell_gradient.max(axis=1).toarray().flatten()
```

Comment:
```suggestion
        cell_distances = self.cell_gradient.max(axis=1).toarray().ravel()
```
`ravel` will avoid returning a new copy of the array; since a new array is already being created w/ `toarray` this is safe